tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/CodeOwnerEntry.cs

```diff
@@ -12,81 +14,105 @@ internal class CodeOwnerEntry
         public List<string> Owners { get; set; } = new List<string>();
 
         public List<string> Labels { get; set; } = new List<string>();
-
-        public static CodeOwnerEntry Parse(string line)
+        public bool IsValid
         {
-            // The format for the Codeowners File will be:
-            // <path> @<owner> ... @<owner> %<label> ... %<label>
-
-            // get rid of tabs
-            line = line.Replace('\t', ' ');
-            // trim the line to get rid of whitespace before and after.
-            line = line.Trim();
-
-            if (string.IsNullOrEmpty(line) || line.StartsWith('#'))
+            get
             {
-                return null;
+                return !string.IsNullOrEmpty(PathExpression);
             }
+        }
+        private static IEnumerable<string> SplitLine(string line)
+        {
+            // Split the line into segments that are delimited by '@', '%' and the end of the string.
 
-            List<string> entries = SplitLine(line).ToList();
-
-            if (!entries.Any())
+            int previousSplit = -1;
+            for (int i = 0; i < line.Length; i++)
             {
-                return null;
+                if (line[i] == '@' || line[i] == '%')
```

Comment:
What splitting on `@` here? Can't we just use the normal .NET split on `%` char?

Reply:
This method is used to split both the owners that use `@` as well as the labels which use `%`.

Given that we will only have one or the other in the new scheme I think using the regular split method will work just fine now. I'll make this change.